### DISCUSSION TITLE
Example of how to generate JSON Schemas programmatically

### DISCUSSION THREAD

**Discussion by tacman** [api-platform/api-platform#2623] [Q&A]
The documentation is a bit sparse on generating JSON Schemas programmatically,

> To generate JSON Schemas programmatically, use the api_platform.json_schema.schema_factory [service](https://symfony.com/doc/current/service_container.html#fetching-and-using-services).

In particular, I want to build a schema from data, not from an existing class.  That is, I'm trying to generate a schema to give some structure to my json blobs.  

So I'm looking for an example that is something like 

```php
function generateSchemaController(
        SchemaFactoryInterface $schemaFactory
        ): Response
{
   // this is fictional, I'm not at all sure what it's supposed to be.
    $property = new Property('age', type: 'integer', 'minimum' => 18);
    $schemaFactory->addProperty($property);

   $json = $schemaFactor->generate();
   return new JsonResponse($json);
}
```

I see that there are classes for Property and Definition, but I'm not sure how to use them to generate a schema.  

I've looked at a few other tools, but ApiPlatform is much more complete, I think an example that doesn't require a PHP class would help a lot.  Thanks.

**Comment by Renrhaf** [comment]
Interested, I need to have an operation which can accept both an entity or an array of entities.
By using "input: false" on the resource I'm able to make it work, but I would like to generate the OpenAPI schema for property describing the API. I'm wondering how to generate the schema for an entity given a particular serialization group as well.
